lib/archethic/mining/pending_transaction_validation.ex

```diff
@@ -671,8 +671,71 @@ defmodule Archethic.Mining.PendingTransactionValidation do
     end
   end
 
+  @code_max_size Application.compile_env!(:archethic, :transaction_data_code_max_size)
+  defp do_accept_transaction(%Transaction{type: :contract, data: %TransactionData{code: code}}, _) do
+    cond do
+      byte_size(code) == 0 ->
+        {:error, "invalid contract type transaction -  code is empty "}
+
+      byte_size(code) >= @code_max_size ->
+        {:error, "invalid contract type transaction , code exceed max size "}
+
+      true ->
+        :ok
+    end
+  end
+
+  @content_max_size Application.compile_env!(:archethic, :transaction_data_content_max_size)
+  @ownership_max_keys Application.compile_env!(:archethic, :ownership_max_authorized_keys)
+  @max_ownerships @ownership_max_keys
+  defp do_accept_transaction(
+         %Transaction{
+           type: :data,
+           data: %TransactionData{content: content, ownerships: ownerships}
+         },
+         _
+       ) do
+    # content_size = byte_size(content) already handled
+    nb_ownerships = length(ownerships)
+
+    cond do
+      content == "" && nb_ownerships == 0 ->
+        {:error, "invalid data type transaction - Both content & ownership are empty"}
+
+      nb_ownerships > @max_ownerships ->
+        {:error, "invalid data type transaction - ownerships exceeds limit"}
+
+      nb_ownerships != 0 ->
+        validate_ownerships(ownerships)
```

Comment:
we may take this opportunity to `validate_ownerships` on all tx types that use it (= a lot of them)

Reply:
To check the size, we can check the overall transaction size < limit

Reply:
> To check the size, we can check the overall transaction size < limit

It will break the contract, as we have split aeweb txs in the size of 3.1 for the content, and the size of the rest of the fields  are not accounted yet.

So we have only set max size for tx content not the whole tx.

@samuelmanzanera

Reply:
In the aeweb CLI the 3MB split is only for the content part. Hence the CLI is only adding information in the content part. 
This should be fine.

Reply:
```// 3_145_728 represent the maximum size for a transaction
// 45_728 represent json tree size
const MAX_FILE_SIZE = 3_145_728 - 45_728
const AEWEB_VERSION = 1``` ?